kubric/core/base.py

```diff
@@ -24,49 +26,65 @@
 
 
 def next_global_count(name, reset=False):
-  """ Return the total number of times this function has been called with the given name.
+  """ Return the total number of times (0-indexed) the function has been called with the given name.
    Used to create the increasing UID counts for each class (e.g. "Sphere.007").
-   When passing reset=True, then all counts are reset to 0.
+   When passing reset=True, then all counts are reset.
    """
   if reset or not hasattr(next_global_count, "counter"):
     next_global_count.counter = collections.defaultdict(int)
     next_global_count.lock = multiprocessing.Lock()
 
   with next_global_count.lock:
+    counter = next_global_count.counter[name]
     next_global_count.counter[name] += 1
-    return next_global_count.counter[name]
+    return counter
 
 
 class Asset(tl.HasTraits):
   """ Base class for the entire OO interface in Kubric.
   All objects, materials, lights, and cameras inherit from Asset.
 
-  Assets
-   * have a UID
-   * are hashable,
-   * have traits that can be observed (by external objects)
-   * support inserting keyframes for certain traits
-   * track linked (external) objects
+  An assets have a UID, is hashable, have traits that can be observed (by external objects), 
+  support inserting keyframes for certain traits, and track linked (external) objects.
+
+  Traits:
+    name: The basename used to create uids (defaults to __class__.name).
+    uid: A unique identifier (e.g. "Cube.001", "Cube.002" , ...)
+    background: TODO.
+    metadata: TODO.
+
+  Attributes:
+    scenes: TODO.
+    keyframes: TODO.
+    linked_objects: TODO.
   """
 
+  name = tl.Unicode(read_only=True)
   uid = tl.Unicode(read_only=True)
   background = tl.Bool(default_value=False)
   metadata = tl.Dict(key_trait=tl.ObjectName())
 
   def __init__(self, **kwargs):
+    # --- ensure all constructor arguments are used
+    # TODO(klaus): this logic could be moved to util.py? (quite generic) 
```

Comment:
I don't see any advantage from moving since we don't need it anywhere else (because every asset inherits from Asset).

Reply:
No, I meant that it feels more like quite a generic feature for traitlets, more than being specific to what we do.

Reply:
Ok, I made it into a one-liner (much easier to see).
Makes no sense to isolate a one-liner :)